docs(samples): add Agent Assist code samples

rogers140 opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-dialogflow/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

engelke commented:
This is a very big PR, well over 1000 lines of new code. It's going to be a while before I can get the time to review it.

rogers140 commented:
> This is a very big PR, well over 1000 lines of new code. It's going to be a while before I can get the time to review it.

Sorry about that. We are actually just migrating private code samples to public for Agent Assist Public Preview Launch. If there is a better way to do this, please let me know.

engelke commented on samples/snippets/participant_management.py:
```diff
@@ -0,0 +1,188 @@
+    Args:
+        participant_name: resource name of the participant.
+        sample_rate_herz: herz rate of the sample.
+        audio_generator: a sequence of audio data.
```
This doesn't match the argument list above.

rogers140 replied:
Fixed.

engelke reviewed (changes requested):
Pass lint and all tests.
Make sure all functions have tests.
Update copyright dates if anything has been written or first published this year.
Have white space before and after import statements and internally to break code into logical chunks for readability.
Use \n in print statements only when double spacing is needed, and use consistently.

rogers140 commented on samples/snippets/answer_record_management.py:
```diff
@@ -0,0 +1,46 @@
+def update_anwer_record(project_id, answer_record_id, click):
+    """Update the answer record.
+
+    Args:
```
Added comments.

rogers140 commented on samples/snippets/conversation_management.py:
```diff
@@ -0,0 +1,146 @@
+    client = dialogflow.ConversationsClient()
+    conversation_path = client.conversation_path(project_id, conversation_id)
+
+    response = client.get_conversation(name=conversation_path)
```
I checked existing code and it seems we don't quite focus on the failure handling.

rogers140 commented on samples/snippets/knowledge_base_management.py:
```diff
@@ -19,12 +19,37 @@
 import argparse
 
 
+# [START dialogflow_list_knowledge_base]
```
Removed this function.

rogers140 commented on samples/snippets/analyze_content_text_test.py:
```diff
@@ -0,0 +1,144 @@
+# Copyright 2019 Google LLC
```
Fixed.

rogers140 commented on samples/snippets/answer_record_management.py:
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google LLC
```
Fixed.

rogers140 commented on samples/snippets/answer_record_management.py:
```diff
@@ -0,0 +1,46 @@
+
+
+# [START dialogflow_update_answer_record]
+def update_anwer_record(project_id, answer_record_id, click):
```
Fixed.

rogers140 commented on samples/snippets/answer_record_management.py:
```diff
@@ -0,0 +1,46 @@
+        project_id: The GCP project linked with the conversation profile.
+        answer_record_id: The answer record id returned along with the
+        suggestion."""
+    from google.cloud import dialogflow_v2 as dialogflow
```
Fixed.

rogers140 commented on samples/snippets/participant_management.py:
```diff
@@ -0,0 +1,188 @@
+    print('Reply Text: {}'.format(response.reply_text))
+
+    for suggestion_result in response.human_agent_suggestion_results:
+        if suggestion_result.error is None:
```
Fixed

rogers140 commented:
> Pass lint and all tests.
> Make sure all functions have tests.
> Update copyright dates if anything has been written or first published this year.
> Have white space before and after import statements and internally to break code into logical chunks for readability.
> Use \n in print statements only when double spacing is needed, and use consistently.

Fixed the issue you mentioned and better organized the tests. Removed some unnecessary functions and samples.

I think I don't know how to update the readme.rst, it seems autogenerated by readme.in but I don't know how it works. Do you know?
